### ISSUE TITLE
Sending SMS via SIM1 stops working

### ISSUE THREAD

**Issue by denknl** [signalapp/Signal-Android#12250]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

**Bug description**
Sending SMS works fine for 1 or 2 days but suddenly it stops working throug SIM1. "Message could not be sent. Check your connection and try again". Connection and network coverage is OK. SMS via WiFi is turned off. Sending SMS via SIM2 (same provider) still works. Receiving SMS works fine either. The workaround to get it back working is to reboot the phone. Switching back to the built-in SMS functionality is another workaround, but I prefer to use Signal for SMS.

New SIM card, flight mode toggle and forcing app to stop, do not work as a solution. I'm experiencing this problem for 6 weeks now, since I switched phones. My previous phone (Huawei P10) didn't have the problem with the same SIM cards.

I really tried to find the behaviour that might be the trigger for sending SMS to stop working, but couldn't figure it out myself.

**Steps to reproduce**
Open Signal
Open chat
Write message
Change send button to "Insecure SMS SIM1"
Send message

Expected result: Message should be sent within 5 seconds.

**Screenshots**
![signal-2022-05-16-142549_001](https://user-images.githubusercontent.com/23361603/168592065-a2f3bf29-6ab3-4cf6-901b-662a6e369971.jpeg)

**Device info**
Device: Xiaomi 12 (2201123G)
Android version: 12 (MIUI 13.0.10)
Signal version: 5.36.3 & 5.37.4

**Link to debug log**
https://debuglogs.org/android/5.36.3/38bb3a0b6e2101be01a527ba990acad47e67fae387666789a7790554184bec44
https://debuglogs.org/android/5.37.4/fdc2619a98be4d2260ed4af09c01cab4ac442423bdda162b1998c5dff5118fc0
In both files, the last message is an example message which is not sent.

**Comment by denknl** [comment]
Yesterday I experienced something interesting. I've sent a test SMS message succesful via SIM1 at 22:53:49. Right after that, I received a Signal message. Right after that, I tried to send a test SMS message via SIM1 again, and it failed. That was about 1-2 minutes later. I immediately gathered the logs:
https://debuglogs.org/android/5.37.4/74c025dffff82fd351075d8fa1178206fdb4fa51f7a985e691b228ebe98601b3
